EpsilonDriverControls/Tests/tests/DriverControlsTest/DriverControlsTest.c

```diff
@@ -0,0 +1,1005 @@
+    }
+
+    HAL_GPIO_TogglePin_Expect(LED_GREEN_GPIO_Port, LED_GREEN_Pin);
+
+    CanMsg expectedMotorDriveMsg = {};
+    osPoolAlloc_ExpectAndReturn(canPool, &expectedMotorDriveMsg);
+    osMessagePut_ExpectAndReturn(canQueue, (uint32_t)&expectedMotorDriveMsg, osWaitForever, osOK);
+
+    CanMsg expectedMotorPowerMsg = {};
+    osPoolAlloc_ExpectAndReturn(canPool, &expectedMotorPowerMsg);
+    osMessagePut_ExpectAndReturn(canQueue, (uint32_t)&expectedMotorPowerMsg, osWaitForever, osOK);
+
+    HAL_GPIO_ReadPin_ExpectAndReturn(RESET_GPIO_Port, RESET_Pin, 1);
+    sendDriveCommands(&prevWakeTime, &expectedDriveCommandsInfo);
+
+    checkDriveMsgStdidAndDlcForDriveCommands(&expectedMotorDriveMsg);
+    checkPowerMessageForDriveCommands(&expectedMotorPowerMsg);
+
+    TEST_ASSERT_EQUAL_MESSAGE(0, expectedDriveCommandsInfo.prevResetStatus,
+                              "Expected driveCommandsInfo.prevResetStatus not 1");
+    TEST_ASSERT_EQUAL_MESSAGE(0, expectedDriveCommandsInfo.motorCurrentOut,
+                              "Expected driveCommandsInfo.motorCurrentOut not 0");
+    TEST_ASSERT_EQUAL_MESSAGE(2, expectedDriveCommandsInfo.accelQueueIndex,
+                              "Expected driveCommandsInfo.accelQueueIndex not 2");
+
+    //memcpy copies 8 bytes of data into msg.data and msg.data stores 8 bits of data in each index.
+    // MAX_REVERSE_RPM = -20000
+    //driveCommandsInfo->motorCurrentOut = 0.05741667
+    uint8_t expectedMsgData[8] = {0x00, 0x40, 0x9C, 0xC6, 0xBE, 0x2D, 0x6B, 0x3D};
+    TEST_ASSERT_EQUAL_UINT8_ARRAY_MESSAGE(expectedMsgData, expectedMotorDriveMsg.Data, 8,
+                                          "Expected motor drive msg.Data context or size incorrect");
+}
+
+//Test sendDriveCommands when regen, brake, forward and reverse are all false.
+//Check that motorCurrentOut == 0 and motorVelocityOut == 0s
+void test_sendDriveCommands_regenBrakeForwardAndReverseOff()
+{
+    DriveCommandsInfo expectedDriveCommandsInfo =
+    {
+        .motorCurrentOut = 0.0f,
+        .prevResetStatus = 0,
+        .regenQueueIndex = 0,
+        .accelQueueIndex = 0,
+    };
+
+    uint32_t prevWakeTime = 0;
+    osDelayUntil_ExpectAndReturn(&prevWakeTime, DRIVE_COMMANDS_FREQ, 0);
+    HAL_ADC_PollForConversion_ExpectAndReturn(&hadc1, ADC_POLL_TIMEOUT, HAL_ERROR);
+    HAL_ADC_PollForConversion_ExpectAndReturn(&hadc2, ADC_POLL_TIMEOUT, HAL_ERROR);
+
+    HAL_GPIO_ReadPin_ExpectAndReturn(FORWARD_GPIO_Port, FORWARD_Pin, 1);
+    HAL_GPIO_ReadPin_ExpectAndReturn(REVERSE_GPIO_Port, REVERSE_Pin, 1);
+    HAL_GPIO_ReadPin_ExpectAndReturn(BRAKES_GPIO_Port, BRAKES_Pin, 1);
+
+    motor0VehicleVelocityInput = 0.0f;
+    motor1VehicleVelocityInput = 0.0f;
+
+    for (int i = 0; i < REGEN_QUEUE_SIZE; i++)
+    {
+        regenValuesQueue[i] = 0;
+    }
+
+    for (int i = 0; i < ACCEL_QUEUE_SIZE; i++)
+    {
+        accelValuesQueue[i] = 0;
+    }
+
+    HAL_GPIO_TogglePin_Expect(LED_BLUE_GPIO_Port, LED_BLUE_Pin);
```

Comment:
This causes an error if you put this test before `reverseIsTrueAndAccelpercentageGreaterThanNonZeroThreshold` - the main drivers code only toggles the blue LED inside forward, so this shouldn't be in this test.